src/glue/jobs/s3_to_json.py

```diff
@@ -402,6 +447,22 @@ def write_file_to_json_dataset(
                 delete_upon_successful_upload=delete_upon_successful_upload
         )
         uploaded_files.append(current_output_path)
+        logger.info(
+                "Output file attributes",
+                extra={
+                    **logger_context,
+                    "file.LineCount": line_count,
+                    "event.kind": "metric",
+                    "event.category": ["file"],
+                    "event.type": ["info", "creation"],
+                    "event.action": "list-file-properties",
+                    "labels": {
+                        k: v.isoformat()
+                        if isinstance(v, datetime.datetime) else v
+                        for k, v in metadata.items()
+                    }
+                }
+    )
```

Comment:
Should these comments be actual comments in the code?

Reply:
+1

Reply:
Maybe? I was taught a lightweight commenting philosophy: Use the docstring to describe the behavior of code and only use inline comments when there is particularly confusing logic. Otherwise, let the code speak for itself.

In this specific case, is it helpful to have an inline comment saying that this log message includes the line count when the line:
```
 "file.LineCount": line_count
```
is there in plain sight? I think making note of the line count log makes more sense in the docstring.

Reply:
This function is cognitively complex (Branching paths, nested for loops, and some setup before it gets there). The docstring does a great job to explain what the function is doing, but in-line comments give a great how perspective.
```

                for transformed_record in transformed_block:
                    # Sum total line count of the file by counting lines in each individual block
                    line_count += 1
```

On a side note after reading a lot of code I'm trying to push towards the "Never Nesting" methodology personally where I can. https://www.youtube.com/watch?v=CFRhGnuXG-4 is a great video I liked on the topic

Reply:
> On a side note after reading a lot of code I'm trying to push towards the "Never Nesting" methodology

I like this! I often have this complexity problem.

I'm going to refactor some of the code now. Maybe I can get it to the point where in-line comments are effectively redundant with the function calls but if not I can add inline comments.

Reply:
I was able to pull out some of the logic into its own function, but most of the code has to do with setup/initialization and so I've added inline comments to point out what each section is doing.